### PR TITLE
fix: enhance isTablet() on Android

### PR DESCRIPTION
**Pull request by mikpalm**
Changed the nature of IsTablet for Android, since newer devices with higher dp and screensizes returns as tablet, needs further testing.

### CODE REVIEW

**Comment by Ingibjorg** [comment]
@mikpalm  and @machour how is this coming along? The issue is tagged with "help wanted". How can I help?

**Comment by machour** [comment]
@Ingibjorg we need more testing as mentioned by Mikael.
Do you have a failing isTablet() test that is fixed by this PR?

**Comment by mikpalmen** [comment]
Can be tested on almost any newer android. Change the resolution from standard to high. Then logout the response from isTablet. it will give you true. when this fix is not applied. I have only tried it on 2 android phones but on Emulators for tablets and more phones.

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 1]
```diff
@@ -90,8 +90,21 @@ private Boolean isEmulator() {
...
+    if (layout) {
+      DisplayMetrics metrics = new DisplayMetrics();
+      getCurrentActivity().getWindowManager().getDefaultDisplay().getMetrics(metrics);
+      if (metrics.densityDpi == DisplayMetrics.DENSITY_DEFAULT
+              || metrics.densityDpi == DisplayMetrics.DENSITY_HIGH
+              || metrics.densityDpi == DisplayMetrics.DENSITY_MEDIUM
+              || metrics.densityDpi == DisplayMetrics.DENSITY_TV
+              || metrics.densityDpi == DisplayMetrics.DENSITY_XHIGH) {
+        return true;
+      } else {
```
Comment (machour) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: The else can be dropped, `false` is returned at the end of the function

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 2]
```diff
@@ -90,8 +90,21 @@ private Boolean isEmulator() {
   }
 
   private Boolean isTablet() {
-    int layout = getReactApplicationContext().getResources().getConfiguration().screenLayout & Configuration.SCREENLAYOUT_SIZE_MASK;
-    return layout == Configuration.SCREENLAYOUT_SIZE_LARGE || layout == Configuration.SCREENLAYOUT_SIZE_XLARGE;
+    boolean layout = (getReactApplicationContext().getResources().getConfiguration().screenLayout & Configuration.SCREENLAYOUT_SIZE_MASK) >= Configuration.SCREENLAYOUT_SIZE_LARGE;
```
Comment (machour) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: Even if `Configuration.SCREENLAYOUT_SIZE_LARGE` is 3 and `SCREENLAYOUT_SIZE_XLARGE;` is 4, I wouldn't rely on that as we never know what may be added in the future..

Can you switch back to testing both values? 🙏

**Comment by machour** [comment]
Ew, sorry I thought that `needs further testing` was referring to this PR :)

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java (hunk 2 above)
Comment (mikpalmen) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: Removed the unnecessary else. What do you meen by switching back to test both. Im doing bigger than or equal?
```
 >= Configuration.SCREENLAYOUT_SIZE_LARGE;
```

Comment (machour) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]:
```
int layout = getReactApplicationContext().getResources().getConfiguration().screenLayout & Configuration.SCREENLAYOUT_SIZE_MASK;

if (layout != Configuration.SCREENLAYOUT_SIZE_LARGE && layout != Configuration.SCREENLAYOUT_SIZE_XLARGE) {
  return false;
}

// rest of your code, without the if (layout)
```

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 3]
```diff
@@ -90,8 +90,19 @@ private Boolean isEmulator() {
...
+    if (layout) {
+      DisplayMetrics metrics = new DisplayMetrics();
```
Comment (Ingibjorg) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: I got a build error on this line: `error: cannot find symbol DisplayMetrics`. Did you forget to import android.util.DisplayMetrics?

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java (hunk 2 above)
Comment (mikpalmen) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: done

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java (hunk 3 above)
Comment (machour) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: Nice catch!

Comment (mikpalmen) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: Yes fixing.

**Comment by machour** [comment]
Thank you both 🙌

I'll test this PR tomorrow and will ship it in 0.17.2

**Pull request merged**